language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ModulePartNodeContextUtil.java

```diff
@@ -62,22 +69,63 @@ private ModulePartNodeContextUtil() {
     public static List<LSCompletionItem> getTopLevelItems(BallerinaCompletionContext context) {
         ArrayList<LSCompletionItem> completionItems = new ArrayList<>();
         // Here we should add the function keyword as well, although it is added via #getTypeItems
-        List<Snippet> snippets = Arrays.asList(
-                Snippet.KW_IMPORT, Snippet.KW_TYPE, Snippet.KW_PUBLIC, Snippet.KW_ISOLATED,
+        List<Snippet> snippets = new ArrayList<>(Arrays.asList(Snippet.KW_TYPE, Snippet.KW_PUBLIC, Snippet.KW_ISOLATED,
                 Snippet.KW_FINAL, Snippet.KW_CONST, Snippet.KW_LISTENER, Snippet.KW_CLIENT, Snippet.KW_VAR,
                 Snippet.KW_ENUM, Snippet.KW_XMLNS, Snippet.KW_CLASS, Snippet.KW_TRANSACTIONAL,
-                Snippet.DEF_FUNCTION, Snippet.DEF_EXPRESSION_BODIED_FUNCTION, Snippet.DEF_MAIN_FUNCTION,
-                Snippet.KW_CONFIGURABLE, Snippet.DEF_ANNOTATION, Snippet.DEF_RECORD, Snippet.STMT_NAMESPACE_DECLARATION,
+                Snippet.DEF_FUNCTION, Snippet.DEF_EXPRESSION_BODIED_FUNCTION, Snippet.KW_CONFIGURABLE,
+                Snippet.DEF_ANNOTATION, Snippet.DEF_RECORD, Snippet.STMT_NAMESPACE_DECLARATION,
                 Snippet.DEF_OBJECT_SNIPPET, Snippet.DEF_CLASS, Snippet.DEF_ENUM, Snippet.DEF_CLOSED_RECORD,
                 Snippet.DEF_ERROR_TYPE, Snippet.DEF_TABLE_TYPE_DESC, Snippet.DEF_TABLE_WITH_KEY_TYPE_DESC,
                 Snippet.DEF_STREAM, Snippet.DEF_SERVICE_COMMON, Snippet.DEF_CLIENT_DECLARATION
-        );
-
+        ));
+        //Add import keyword conditionally
+        if (isInImportStatementsContext(context)) {
+            snippets.add(Snippet.KW_IMPORT);
+        }
+        if (!isMainFunctionAvailable(context)) {
+            snippets.add(Snippet.DEF_MAIN_FUNCTION);
+        }
         snippets.forEach(snippet -> completionItems.add(new SnippetCompletionItem(context, snippet.get())));
 
         return completionItems;
     }
 
+    private static boolean isMainFunctionAvailable(BallerinaCompletionContext context) {
+        Optional<ModulePartNode> modulePartNode = getModulePartNode(context);
+        if (modulePartNode.isEmpty()) {
+            return false;
+        }
+        return modulePartNode.get().members().stream().anyMatch(moduleMemberDeclarationNode ->
```

Comment:
I think we should be able to use `visibleSymbols` to determine if we have a `main()` function already. Shall we use that approach?